optuna/study.py

```diff
@@ -417,6 +417,10 @@ def ask(self) -> trial_module.Trial:
         trial_id = self._pop_waiting_trial_id()
         if trial_id is None:
             trial_id = self._storage.create_new_trial(self._study_id)
+
+        # update the trial state to running and **change the datetime_start**
+        self._storage.set_trial_state(trial_id, TrialState.RUNNING)
```

Comment:
Does the trial already have the `RUNNING` state and a non-Null  `datetime_start` before reaching those lines?

Reply:
Yeah when the trial is created directly this leads to repeated changes to trial state. Already fixed in the latest commit. So now only those waiting trials will be updated.